Troubleshooting Guide for mixed versions in Java / Android SDK (#13140)
getsentry/sentry-docs: docs/platforms/android/troubleshooting/mixed-versions/index.mdx

```groovy {tabTitle:Gradle}{filename:build.gradle}
implementation('io.sentry:sentry:8.6.0')
implementation('io.sentry:sentry-android:8.7.0')
```
```xml {tabTitle:Maven}{filename:pom.xml}
<dependencyManagement>
    <dependencies>
        <dependency>
            <groupId>io.sentry</groupId>
            <artifactId>sentry</artifactId>
            <version>8.6.0</version>
        </dependency>
        <dependency>
            <groupId>io.sentry</groupId>
            <artifactId>sentry-android</artifactId>
            <version>8.7.0</version>
        </dependency>
    </dependencies>
</dependencyManagement>
```

## Build Plugin And Explicit Dependency

When using our Gradle or Maven plugin and manually defining additional Sentry Android SDK dependencies it is also possible to end up with mixed versions. The following snippet shows the plugin being configured to use version `8.0.0` but there is also an additional dependency that has been set to version `8.1.0`. To fix the issue please set the same version or use `sentry-bom`.

> ```groovy {tabTitle:Gradle}{filename:build.gradle}

Reviewer: We could also add the snippet to add the plugin e.g.
```
plugins {
    id "io.sentry.android.gradle" version "5.3.0"
}
```

> ```groovy {tabTitle:Gradle}{filename:build.gradle}

Author: ```suggestion
```groovy {tabTitle:Gradle}{filename:build.gradle}
plugins { id "io.sentry.android.gradle" version "5.3.0" }

```

---
title: Mixed Versions
description: "Troubleshoot and resolve mixed Android SDK dependency versions."
sidebar_order: 1000
---

> Using multiple Android SDK dependencies with mixed versions is not supported as it is very likely this leads to a crash later on. For this reason we chose to crash the application on SDK init instead.

Reviewer 2: ```suggestion
Using multiple Android SDK dependencies with mixed versions is not supported as it is very likely to lead to a crash later on. For this reason we chose to crash the application on SDK init instead.
```

## Multiple Dependencies With Mixed Versions

> The following snippet shows a mix of version caused by using `sentry` with version `8.6.0` and `sentry-android` with version `8.7.0`. To fix the issue please set the same version for all dependencies or <PlatformLink to="/configuration/bill-of-materials/">use `sentry-bom`</PlatformLink>. This may also happen if you are using an internal library which has a different version defined.

Reviewer 2: ```suggestion
The following snippet shows a mixed version conflict caused by using `sentry` with version `8.6.0` and `sentry-android` with version `8.7.0`. To fix the issue, set the same version for all dependencies or <PlatformLink to="/configuration/bill-of-materials/">use `sentry-bom`</PlatformLink>. This may also happen if you are using an internal library which has a different version defined.
```

```groovy {tabTitle:Gradle}{filename:build.gradle}
implementation('io.sentry:sentry:8.6.0')
implementation('io.sentry:sentry-android:8.7.0')
```
```xml {tabTitle:Maven}{filename:pom.xml}
<dependencyManagement>
    <dependencies>
        <dependency>
            <groupId>io.sentry</groupId>
            <artifactId>sentry</artifactId>
            <version>8.6.0</version>
        </dependency>
        <dependency>
            <groupId>io.sentry</groupId>
            <artifactId>sentry-android</artifactId>
            <version>8.7.0</version>
        </dependency>
    </dependencies>
</dependencyManagement>
```

## Build Plugin And Explicit Dependency

> When using our Gradle or Maven plugin and manually defining additional Sentry Android SDK dependencies it is also possible to end up with mixed versions. The following snippet shows the plugin being configured to use version `8.0.0` but there is also an additional dependency that has been set to version `8.1.0`. To fix the issue please set the same version or <PlatformLink to="/configuration/bill-of-materials/">use `sentry-bom`</PlatformLink>.

Reviewer 2: ```suggestion
When using our Gradle or Maven plugin and manually defining additional Sentry Android SDK dependencies, it is also possible to end up with mixed versions. The following snippet shows the plugin being configured to use version `8.0.0` but there is an additional dependency that has been set to version `8.1.0`. To fix the issue,  set the same version or <PlatformLink to="/configuration/bill-of-materials/">use `sentry-bom`</PlatformLink>.
```
